File: src/frontend/src/types/matriculas.ts
import type { Grado } from './grados';

export interface AlumnoMin {
  id: number;
  username: string;
  nombre: string;
  apellidos: string;
  email: string;
}

export interface ResponsableMin {
  id: number;
  nombre: string;
  apellidos: string;
}

export interface AsignaturaCatalogoEmbed {
  id: number;
  codigo: string;
  nombre: string;
  ects: number;
  caracter: 'OB' | 'OP' | 'FB';
  curso_plan: number;
}

export interface AsignaturaMatriculadaDetalle {
  id: number;
  n_matricula: number;
  asignatura: AsignaturaCatalogoEmbed;
}

export interface MatriculaDetalle {
  id: number;
  alumno: AlumnoMin;
  curso_academico: string;
  fecha_importacion: string;
  responsable: ResponsableMin;
  grado: Grado;
  asignaturas_matriculadas: AsignaturaMatriculadaDetalle[];
}

export interface MatriculaListaItem {
  id: number;
  alumno: AlumnoMin;
  curso_academico: string;
  grado: Grado;
  fecha_importacion: string;
  num_asignaturas: number;
}

File: src/backend/app/schemas/matriculas.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.schemas.grados import GradoOut


class AsignaturaCatalogoEmbedOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    codigo: str
    nombre: str
    ects: float
    caracter: str
    curso_plan: int


class AsignaturaMatriculadaDetalleOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    n_matricula: int
    asignatura: AsignaturaCatalogoEmbedOut


class AlumnoMinOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    username: str
    nombre: str
    apellidos: str
    email: str


class ResponsableMinOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    nombre: str
    apellidos: str


class MatriculaDetalleOut(BaseModel):
    """Agregado completo de matrícula con sus asignaturas."""

    id: int
    alumno: AlumnoMinOut
    curso_academico: str
    fecha_importacion: datetime
    responsable: ResponsableMinOut
    grado: GradoOut
    asignaturas_matriculadas: list[AsignaturaMatriculadaDetalleOut]


class MatriculaListaItemOut(BaseModel):
    id: int
    alumno: AlumnoMinOut
    curso_academico: str
    grado: GradoOut
    fecha_importacion: datetime
    num_asignaturas: int
